workflow-manager/main.go

```diff
@@ -420,16 +420,27 @@ func (b *bucket) listFilesGS(ctx context.Context) ([]string, error) {
 	log.Printf("looking for ready batches in gs://%s as (ambient service account)", b.bucketName)
 	var output []string
 	it := bkt.Objects(ctx, query)
+
+	// https://cloud.google.com/storage/docs/json_api/v1/objects/list
+	// We can have up to: 1000
```

Comment:
I appreciate the link to the API docs! However the comment is a bit cryptic. 1000 what? Also the value we passed below is 100, so they seem disconnected. I'm assuming you used 100 to mirror what we do in the S3 case, which is a good idea, but this comment should explain that. How about:

```
Use the paginated API to list bucket contents, as otherwise we would only get the first 1,000 objects in the bucket. As in the S3 case above, we get 100 results at a time to ensure we exercise the pagination case.
https://cloud.google.com/storage/docs/json_api/v1/objects/list
```